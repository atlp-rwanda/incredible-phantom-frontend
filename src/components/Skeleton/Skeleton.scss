.profile-wrapper {
  width: 30%;
  .profile {
    width: 100%;
    margin: 10px 0px 50px 10px;
  }
}
