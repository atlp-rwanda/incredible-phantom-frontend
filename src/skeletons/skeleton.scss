.skeleton {
  background-color: #ddd;
  margin: 10px 0;
}

.skeleton-wrapper {
  margin: 5px auto;
  padding: 5px 10px;
}
.skeleton.title {
  width: 50%;
  height: 40px;
}
.skeleton.text {
  width: 40%;
  height: 12px;
  margin-bottom: 10px;
}

.skeleton.nav {
  width: 100%;
  height: 8vh;
  margin-bottom: 50px;
}
.skeleton.body {
  width: 40%;
  height: 40vh;
  margin-bottom: 50px;
}
.skeleton.button {
  margin-top: 30px;
  width: 15%;
  height: 50px;
}
.skeleton.avatar {
  width: 100px;
  height: 100px;
  border-radius: 50%;
}
.skeleton.thumbnail {
  width: 100px;
  height: 100px;
}
.skeleton-profile {
  display: grid;
  grid-template-columns: 1fr 2fr;
  gap: 1px;
  align-items: center;
}
.skeleton-landing {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
