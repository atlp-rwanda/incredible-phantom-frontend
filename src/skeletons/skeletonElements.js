import React from 'react';
import './skeleton.scss';

export const SkeletonElements = ({ type }) => {
  
  const classes = `skeleton ${type}`;

  return <div className={classes}></div>;
};
