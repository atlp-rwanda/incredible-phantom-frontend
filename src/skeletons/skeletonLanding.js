import React from 'react';
import { SkeletonElements } from './skeletonElements';
import './skeleton.scss';
export const SkeletonLanding = () => {
  return (
    <div className='skeleton-wrapper'>
      <div className='skeleton-landing'>
        <SkeletonElements type='nav' />
        <SkeletonElements type='title' />
        <SkeletonElements type='text' />
        <SkeletonElements type='body' />
        <SkeletonElements type='button' />
      </div>
    </div>
  );
};
