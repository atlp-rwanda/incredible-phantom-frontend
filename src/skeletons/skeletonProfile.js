import React from 'react';
import { SkeletonElements } from './skeletonElements';
import './skeleton.scss';

export const SkeletonProfile = () => {
  return (
    <div className="skeleton-wrapper">
      <div className="skeleton-profile">
        <div>
          <SkeletonElements type="avatar" />
        </div>
        <div>
          <SkeletonElements type="title" />
          <SkeletonElements type="text" />
          <SkeletonElements type="text" />
        </div>
      </div>
    </div>
  );
};
