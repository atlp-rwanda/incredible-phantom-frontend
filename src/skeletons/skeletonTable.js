import React from 'react';
import { SkeletonElements } from './skeletonElements';
import './skeleton.scss';
export const SkeletonTable = () => {
  return (
    <div className='skeleton-table'>
      <SkeletonElements type='text' />
    </div>
  );
};
