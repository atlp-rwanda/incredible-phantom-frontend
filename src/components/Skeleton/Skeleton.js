import React, { useEffect, useState } from 'react';
import { SkeletonProfile } from '../../skeletons/skeletonProfile';
import './Skeleton.scss';
export const Skeleton = () => {
  const [profile, setProfile] = useState(null);

  useEffect(() => {
    setTimeout(async () => {
      const response = await fetch(
        'https://jsonplaceholder.typicode.com/users/',
      );
      const data = await response.json();
      setProfile(data);
    }, 2000);
  });

  return (
    <div className="profile-wrapper">
      {profile &&
        profile.map((user) => (
          <div className="profile" key={user.id}>
            <h1>{user.username}</h1>
            <p>{user.email}</p>
            <p>{user.address.suite}</p>
          </div>
        ))}
      {!profile &&
        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((elem) => (
          <SkeletonProfile key={elem} />
        ))}
    </div>
  );
};
