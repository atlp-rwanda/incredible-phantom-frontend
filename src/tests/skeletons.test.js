import React from 'react';
import { shallow } from 'enzyme';
import { SkeletonElements } from '../skeletons/skeletonElements';
import { SkeletonLanding } from '../skeletons/skeletonLanding';
import { SkeletonProfile } from '../skeletons/skeletonProfile';
import { Skeleton } from '../components/Skeleton/Skeleton';

describe('Skeletons', () => {
  it('should return the main skeleton div', () => {
    const wrapper = shallow(<SkeletonElements />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return the Landing page skeleton ui', () => {
    const wrapper = shallow(<SkeletonLanding />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return the main skeleton div', () => {
    const wrapper = shallow(<SkeletonProfile />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return the skeleton page', () => {
    const wrapper = shallow(<Skeleton />);
    expect(wrapper).toMatchSnapshot();
  });
});
