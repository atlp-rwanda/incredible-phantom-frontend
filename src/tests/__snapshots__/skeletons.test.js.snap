// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeletons should return the Landing page skeleton ui 1`] = `
<div
  className="skeleton-wrapper"
>
  <div
    className="skeleton-landing"
  >
    <SkeletonElements
      type="nav"
    />
    <SkeletonElements
      type="title"
    />
    <SkeletonElements
      type="text"
    />
    <SkeletonElements
      type="body"
    />
    <SkeletonElements
      type="button"
    />
  </div>
</div>
`;

exports[`Skeletons should return the main skeleton div 1`] = `
<div
  className="skeleton undefined"
/>
`;

exports[`Skeletons should return the main skeleton div 2`] = `
<div
  className="skeleton-wrapper"
>
  <div
    className="skeleton-profile"
  >
    <div>
      <SkeletonElements
        type="avatar"
      />
    </div>
    <div>
      <SkeletonElements
        type="title"
      />
      <SkeletonElements
        type="text"
      />
      <SkeletonElements
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Skeletons should return the skeleton page 1`] = `
<div
  className="profile-wrapper"
>
  <SkeletonProfile
    key="1"
  />
  <SkeletonProfile
    key="2"
  />
  <SkeletonProfile
    key="3"
  />
  <SkeletonProfile
    key="4"
  />
  <SkeletonProfile
    key="5"
  />
  <SkeletonProfile
    key="6"
  />
  <SkeletonProfile
    key="7"
  />
  <SkeletonProfile
    key="8"
  />
  <SkeletonProfile
    key="9"
  />
  <SkeletonProfile
    key="10"
  />
</div>
`;
